CSS module: box alignment (#37566)
mdn/content: files/en-us/web/css/css_box_alignment/index.md

{{CSSRef}}

> The **CSS box alignment** module specifies CSS features that relate to the alignment of boxes within their containers across all CSS. It defines the alignment of the various CSS box layout models, including block layout, table layout, flex layout, and grid layout, creating a consistent method of alignment across all of CSS. This module contains the features of CSS relating to the alignment of boxes within their containers in the various CSS box layout models: block layout, table layout, flex layout, and grid layout.

Reviewer: ```suggestion
The **CSS box alignment** module specifies CSS features relating to the alignment of boxes within their containers. It defines the alignment of the various CSS box layout models including block layout, table layout, flexible box layout (flexbox), and grid layout, creating a consistent alignment method across all of CSS.
```

The module details alignment terminology, enabling alignment properties to be used in multiple layout modules, rather than limited to a particular layout method.

> Alignment is linked to writing modes in that when we align an item we do not consider whether we are aligning it to the physical dimensions of top, right, bottom and left. Instead we describe alignment in terms of the start and end of the particular dimension we are working with. This ensures that alignment works in the same way whichever writing mode the document has.

Reviewer: ```suggestion
Alignment is linked to writing modes in that when we align an item we do not consider whether we are aligning it to the physical dimensions of top, right, bottom, and left. Instead, we describe alignment in terms of the start and end of the particular dimension we are working with. This ensures that alignment works in the same way whichever writing mode the document has.
```

Reviewer: Also, link "writing modes" to https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_writing_modes?

## Reference

### CSS Properties

- {{cssxref("align-content")}}
- {{cssxref("align-items")}}
- {{cssxref("align-self")}}
- {{cssxref("column-gap")}}
- {{cssxref("gap")}}
- {{cssxref("justify-content")}}
- {{cssxref("justify-items")}}
- {{cssxref("justify-self")}}
- {{cssxref("place-content")}}
- {{cssxref("place-items")}}
- {{cssxref("place-self")}}
- {{cssxref("row-gap")}}

### Data types

- {{cssxref("baseline-position")}}
- {{cssxref("content-distribution")}}
- {{cssxref("content-position")}}
- {{cssxref("overflow-position")}}
- {{cssxref("self-position")}}

### Terms and definitions

- {{Glossary("Alignment container")}}
- {{Glossary("Alignment subject")}}
- [Baseline alignment](/en-US/docs/Web/CSS/CSS_box_alignment/box_alignment#baseline_alignment)
- [Distributed alignment](/en-US/docs/Web/CSS/CSS_box_alignment/box_alignment#distributed_alignment)
- {{Glossary("Fallback alignment")}}
- [Positional alignment](/en-US/docs/Web/CSS/CSS_box_alignment/box_alignment#positional_alignment)

## Guides

> As the CSS box alignment properties are implemented differently depending on the specification they interact with, refer to the following pages for each layout type for details of how to use the alignment properties with it:

Reviewer: An issue here is that this paragraph doesn't relate to the first linked guide (the general concepts/overview); it relates to the others, however. You could change this by altering the wording, or perhaps moving the paragraph to below the first guide link?

> - [Box alignment](/en-US/docs/Web/CSS/CSS_box_alignment/Box_alignment)

Reviewer: As I said in an earlier comment, this should probably be called something like "CSS Box alignment overview".
